Description: Apply '-fno-strict-aliasing' option only in uWSGI binary building
Author: Leonid Borisenko <leo.borisenko@gmail.com>
Forwarded: http://lists.unbit.it/pipermail/uwsgi/2010-September/000666.html
Last-Update: 2010-09-14

Index: uwsgi/uwsgiconfig.py
===================================================================
--- uwsgi.orig/uwsgiconfig.py	2011-01-31 21:35:23.000000000 +0200
+++ uwsgi/uwsgiconfig.py	2011-01-31 21:35:37.000000000 +0200
@@ -95,10 +95,6 @@
 
 cflags = ['-O2', '-Wall', '-Werror', '-D_LARGEFILE_SOURCE', '-D_FILE_OFFSET_BITS=64'] + os.environ.get("CFLAGS", "").split()
 
-# add -fno-strict-aliasing only on python2 and gcc < 4.3
-if (sys.version_info[0] == 2) or (gcc_major < 4) or (gcc_major == 4 and gcc_minor < 3):
-	cflags = cflags + ['-fno-strict-aliasing']
-
 if gcc_major >= 4:
 	cflags = cflags + [ '-Wextra', '-Wno-unused-parameter', '-Wno-missing-field-initializers' ]
 
@@ -131,11 +127,16 @@
 
 def build_uwsgi(bin_name):
 	print("*** uWSGI compiling ***")
+	# add -fno-strict-aliasing only on python2 and gcc < 4.3
+	if (sys.version_info[0] == 2) or (gcc_major < 4) or (gcc_major == 4 and gcc_minor < 3):
+		uwsgi_binary_cflags = cflags + ['-fno-strict-aliasing']
+	else:
+		uwsgi_binary_cflags = cflags
 	for file in gcc_list:
 		objfile = file
 		if objfile == 'uwsgi':
 			objfile = 'main'
-		cmdline = "%s -c %s -o %s %s" % (GCC, ' '.join(cflags), objfile + '.o', file + '.c')
+		cmdline = "%s -c %s -o %s %s" % (GCC, ' '.join(uwsgi_binary_cflags), objfile + '.o', file + '.c')
 		print(cmdline)
 		ret = os.system(cmdline)
 		if ret != 0:
